### PR TITLE
chore(atomix): reduce storage buffers

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

We know that in atomix the large read and write buffers are problematic and are double the size of the max message size, but this is not really necessary. In order to reduce the cache misses and improve the performance I reduced the buffer sizes to the max message size + additional headers.

## Benchmark

I run a benchmark over the week end and got good  results.

#### General
base-general

#### Back pressure
base-pressure

#### Latency
base-latency

#### General

reduce-general

#### Back pressure

reduce-backpressure

#### Latency

reduce-latency

## Last 4 days

Comparison between last 4 days:

| Base | Reduced Buffers |
|---------|-----------------------|
| last-4-days-base | last-4-days-reduce-buffers |

## Related issues

related https://github.com/zeebe-io/zeebe/issues/4249

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: atomix/storage/src/test/java/io/atomix/storage/journal/AbstractJournalTest.java [hunk 1]
```diff
@@ -90,8 +92,8 @@ public static Collection primeNumbers() {
         .withDirectory(folder)
         .withNamespace(NAMESPACE)
         .withStorageLevel(storageLevel())
-        .withMaxSegmentSize(maxSegmentSize)
-        .withMaxEntrySize(48)
+        .withMaxSegmentSize(maxSegmentSize * 2)
```
Comment (deepthidevaki) [inline atomix/storage/src/test/java/io/atomix/storage/journal/AbstractJournalTest.java:95]: Why is this required?

**Review by deepthidevaki — COMMENTED** [review]
Some tests are failing. I couldn't figure out what broke. I will take a detailed look later.

File: atomix/storage/src/test/java/io/atomix/storage/journal/AbstractJournalTest.java (hunk 1 above)
Comment (Zelldon) [reply atomix/storage/src/test/java/io/atomix/storage/journal/AbstractJournalTest.java:95]: The segment size is in the tests to small i was not able to write the max msg size😅

Comment (deepthidevaki) [reply atomix/storage/src/test/java/io/atomix/storage/journal/AbstractJournalTest.java:95]: maxSegmentSize is parameterized, right? May be we can adjust it to start from a higher value.

Comment (Zelldon) [reply atomix/storage/src/test/java/io/atomix/storage/journal/AbstractJournalTest.java:95]: Was not able to find it due to to the abstractions in the test classes. If you see it then feel free😃

Comment (Zelldon) [reply atomix/storage/src/test/java/io/atomix/storage/journal/AbstractJournalTest.java:95]: I think they are set somehow as arguments for the test classes

**Comment by deepthidevaki** [comment]
Update: I looked into it, but couldn't find a clean way to fix the tests. The problem is with the tests. And it is not a bug introduced by this change.

**Comment by Zelldon** [comment]
@deepthidevaki should we close this? It seems we want to migrate to mmap files? Maybe the test is worth to add.

**Comment by deepthidevaki** [comment]
Yes. We can close it, unless you want to fix the tests. We also have another issue to improve the journal tests in general.

**Comment by Zelldon** [comment]
@deepthidevaki would you be fine with adding me just the test that we can write a record with max message size?
